src/encoded/static/components/vis_defines.js

```diff
@@ -382,7 +382,31 @@ export class BrowserSelector extends React.Component {
         // Generate the batch hub URL used in batch visualization query strings.
         const parsedLocationHref = url.parse(this.context.location_href);
         const hostName = `${parsedLocationHref.protocol}//${parsedLocationHref.host}`;
-        const batchHubUrl = generateBatchHubUrl(results, hostName);
+        // if we have 'activeFilters' set on mouse development matrix page, append to visualization link
+        let resultsId = results['@id'];
+        if (activeFilters) {
+            activeFilters.forEach((f) => {
+                if (['adult', 'postnatal', 'embryo'].includes(f)) {
+                    const stageTerm = f === 'embryo' ? 'embryonic' : f;
+                    const stageFilter = `&replicates.library.biosample.life_stage=${stageTerm}`;
+                    if (!resultsId.includes(stageFilter)) {
+                        resultsId = `${resultsId}${stageFilter}`;
+                    }
+                } else {
+                    const stageTerm = f.split(' ')[0] === 'embryo' ? 'embryonic' : f.split(' ')[0];
+                    const ageTerm = f.split(' ').slice(1).join(' ');
+                    const stageFilter = `&replicates.library.biosample.life_stage=${stageTerm}`;
+                    if (!resultsId.includes(stageFilter)) {
+                        resultsId = `${resultsId}${stageFilter}`;
+                    }
+                    const ageFilter = `&replicates.library.biosample.age_display=${ageTerm}`;
+                    if (!resultsId.includes(ageFilter)) {
+                        resultsId = `${resultsId}${ageFilter}`;
+                    }
```

Comment:
There’s one potentially outlandish case where I think this could fail, but I’m not sure if it can practically happen. If you go to https://encd-2666-f761e000e-emma.demo.encodedcc.org/mouse-development-matrix/?type=Experiment&status=released&related_series.@type=OrganismDevelopmentSeries&replicates.library.biosample.organism.scientific_name=Mus+musculus&replicates.library.biosample.life_stage=adult&replicates.library.biosample.age_display=8 weeks&replicates.library.biosample.life_stage=postnatal the server replaces the “8 weeks” with “8%20weeks” which arrives in this code in `resultsId`. If `ageTerm` gets the value “8 weeks” then this `includes` test fails, and `resultsId` gets _another_ copy of “&replicates.library.biosample.age_display=” — the existing one with “8%20weeks” and a new one with “8 weeks.” Again, I’m not sure if this is too unrealistic or not, as this was the only way I could trigger that. Might need to use encodeURIComponent which will replace the space with a plus, and hope that people use a plus in the URL. Ben sometimes doesn’t care what happens when people manually enter the URL, so this might not matter. I’ll leave this as a question.